src/components/Modals/ManualDelegate/ManualDelegate.jsx

```diff
@@ -63,6 +63,7 @@ const ManualDelegate = ({ onClose, isOpen, address, tContract, stakedContract })
 
     const handleDelegateMyself = () => {
         setDelegateAddress(address);
+        setIsValidAddress(true)
```

Comment:
Just a thought: could call `checkAddress(address)` here to reuse logic (this version assumes `address` is in fact a valid address, which isn't crazy but could interact with another future bug unexpectedly).

Either way, thanks for fixing this :grin:

Reply:
My first approach was to reuse checkAddress(address) here to re-render the component, however since address is being passed as params to this component and originally being assigned directly from the wallet class from web3onboard I thought it could be a bit redundant to check it. 
https://github.com/threshold-network/threshold-delegate/blob/ed643be68a7a2faec41ee495ea5c17dfc2e09fe6/src/pages/Home/Home.jsx#L38

However, after further assessment, it does seem a brittle solution to just tag it as a valid address, without double confirming it, mainly considering different unexpected scenarios. Therefore I changed in conform with your suggestion: https://github.com/threshold-network/threshold-delegate/pull/5/commits/c9e82833e03046c5b4d31db5f8c6ed6ff522d866

Thanks for the review. @Shadowfiend